Add Bitvis Power Hub integration (#44082)
home-assistant/home-assistant.github.io: source/_integrations/bitvis.markdown

---
title: Bitvis Power Hub

> description: Instructions on how to integrate your Bitvis Power Hub into Home Assistant.

Reviewer: ```suggestion
description: Read real-time electricity data from your smart meter with the Bitvis Power Hub integration for Home Assistant.
```

> ha_release: '2026.3'

Reviewer: ```suggestion
ha_release: 2026.4
```

ha_category:
  - Energy
  - Sensor
ha_platforms:
  - sensor
ha_iot_class: Local Push
ha_config_flow: true
ha_zeroconf: true
ha_codeowners:
  - '@MandusBorjesson'
  - '@simontegelid'
ha_domain: bitvis
ha_integration_type: device
---

> The **Bitvis Power Hub** {% term integration %} reads real-time electricity data from a [Bitvis Power Hub](https://bitvis.se/) device. The Power Hub connects to the HAN port (Home Area Network) on your smart electricity meter and pushes measurements to Home Assistant over your local network using UDP.

Reviewer: ```suggestion
The **Bitvis Power Hub** {% term integration %} reads real-time electricity data from a [Bitvis Power Hub](https://bitvis.se/) device. The Power Hub connects to the <abbr title="Home Area Network">HAN</abbr> port (a standard port on European smart meters) on your smart electricity meter and pushes measurements to Home Assistant over your local network using UDP.
```

Because data is pushed directly from the device, no cloud connection or polling is required.

> 

Reviewer: ```suggestion

## Prerequisites

Before adding the integration, make sure:

- Your Bitvis Power Hub is connected to your local network and configured to send data to the IP address of your Home Assistant host on the UDP port above (default: `58220`).
- Your router or firewall allows UDP traffic from the Power Hub to Home Assistant on that port.

{% note %}
The Bitvis Power Hub and Home Assistant must be on the same network subnet, or your router/firewall must be configured to forward UDP traffic between them.
{% endnote %}

```
Moving the Prereqs section before the actual Configuration section.

{% include integrations/config_flow.md %}

> ## Configuration parameters

Reviewer: ```suggestion
```

{% configuration_basic %}
Host:
  description: The hostname or IP address of your Bitvis Power Hub on your local network.
Port:
  description: The UDP port Home Assistant listens on for data from the device. Defaults to `58220`.
Name:
  description: A friendly name for this device in Home Assistant.
{% endconfiguration_basic %}

> ## Prerequisites
> 
> Before adding the integration, make sure:
> 
> - Your Bitvis Power Hub is connected to your local network and configured to send data to the IP address of your Home Assistant host on the UDP port above (default: `58220`).
> - Your router or firewall allows UDP traffic from the Power Hub to Home Assistant on that port.
> 
> {% note %}
> The Bitvis Power Hub and Home Assistant must be on the same network subnet, or your router/firewall must be configured to forward UDP traffic between them.
> {% endnote %}
> 

Reviewer: ```suggestion
```

## Sensors

The following sensors are provided, depending on the capabilities of your smart meter:

### Power

> | Sensor | Description |

Reviewer: We [prefer lists over tables](https://developers.home-assistant.io/docs/documenting/general-style-guide#tables), see [example](https://github.com/home-assistant/home-assistant.io/blob/current/source/_integrations/_integration_docs_template.markdown?plain=1#L105).

| Active power import | Total active power consumed from the grid (kW) |
| Active power export | Total active power delivered to the grid (kW) |
| Active power import L1/L2/L3 | Per-phase active power consumed (kW) |
| Active power export L1/L2/L3 | Per-phase active power delivered (kW) |
| Reactive power import | Total reactive power consumed (kvar) |
| Reactive power export | Total reactive power delivered (kvar) |
| Reactive power import L1/L2/L3 | Per-phase reactive power consumed (kvar) |
| Reactive power export L1/L2/L3 | Per-phase reactive power delivered (kvar) |

### Voltage and current

| Sensor | Description |
|---|---|
| Voltage L1/L2/L3 | Phase voltages (V) |
| Current L1/L2/L3 | Phase currents (A) |

### Energy

| Sensor | Description |
|---|---|
| Active energy import | Cumulative active energy consumed from the grid (kWh) |
| Active energy export | Cumulative active energy delivered to the grid (kWh) |
| Reactive energy import | Cumulative reactive energy consumed (kvarh) |
| Reactive energy export | Cumulative reactive energy delivered (kvarh) |

## Diagnostic sensors

The following diagnostic sensors are also available:

| Sensor | Description |
|---|---|

> | Uptime | Device uptime (s) |

Reviewer: ```suggestion
| Uptime | Estimated time the device was last started. Timestamp. |
```
Is uptime in seconds still correct? or is it a timestamp now?

> | WiFi signal strength | Received signal strength (dBm) |

Reviewer: ```suggestion
| Wi-Fi signal strength | Received signal strength (dBm) |
```

| Reactive power export | Total reactive power delivered (kvar) |
| Reactive power import L1/L2/L3 | Per-phase reactive power consumed (kvar) |
| Reactive power export L1/L2/L3 | Per-phase reactive power delivered (kvar) |

### Voltage and current

| Sensor | Description |
|---|---|
| Voltage L1/L2/L3 | Phase voltages (V) |
| Current L1/L2/L3 | Phase currents (A) |

### Energy

| Sensor | Description |
|---|---|
| Active energy import | Cumulative active energy consumed from the grid (kWh) |
| Active energy export | Cumulative active energy delivered to the grid (kWh) |
| Reactive energy import | Cumulative reactive energy consumed (kvarh) |
| Reactive energy export | Cumulative reactive energy delivered (kvarh) |

## Diagnostic sensors

The following diagnostic sensors are also available:

| Sensor | Description |
|---|---|
| Uptime | Device uptime (s) |
| WiFi signal strength | Received signal strength (dBm) |
| HAN messages successfully parsed | Counter of successfully parsed HAN port messages |
| HAN buffer overflows | Counter of HAN port buffer overflow events |
| MAC address | Device MAC address |
| Model | Device model name |

> | Software version | Installed firmware version |

Author: I don't understand what you mean, the table only has singular pipe symbols (`|`) at the start of each row. Could you clarify what you mean?
